File: sdk/python/featureflags/context.py
from dataclasses import dataclass, field
from typing import Any

@dataclass(frozen=True)
class EvaluationContext:
    user_id: str
    attributes: dict[str, Any] = field(default_factory=dict)

    def as_api_user(self) -> dict[str, Any]:
        return {
            "id": self.user_id,
            **self.attributes,
        }

File: sdk/javascript/src/context.js
export class EvaluationContext {
  constructor(userId, attributes = {}) {
    if (!userId) {
      throw new Error("userId is required");
    }
    this.userId = String(userId);
    this.attributes = { ...attributes };
  }

  toApiUser() {
    return {
      id: this.userId,
      ...this.attributes,
    };
  }
}
